### PR TITLE
[PLAT-2808] Resolved Orthogonal Vector

### PR DESCRIPTION
**Pull request by MadisonEhlers-Vertex**
## Summary

These changes are to improve the calculated orthogonal vector when choosing one when the angles are parallel/anti-parallel. This was riffed off of some other [c++ code suggested](https://github.com/Vertexvis/graphics3d-lib-cpp/blob/55d4f15aa228b3a40264cc7d9e9ac369418e4c24/include/LinearTransform.hpp#L290-L297) by @mikemccarty-vertex 

This appears to resolve a few edge cases and has sounder logic than the previous implementation which had a few bugs.

## Test Plan

- Test aligning angles
- Test flipping normals

### CODE REVIEW

File: packages/viewer/src/lib/transforms/transformation-delta.ts [hunk 1]
```diff
@@ -1,26 +1,23 @@
...
-  const z = absNorm.z <= absNorm.x && absNorm.z <= absNorm.y ? 1.0 : 0.0;
+  const dx0 = Vector3.create(0, norm.z, -norm.y);
+  const dx1 = Vector3.create(-norm.z, 0, norm.x);
 
-  const vector = Vector3.create(x, y, z);
-  return Vector3.normalize(Vector3.cross(normal, vector));
+  const mag2Dx0 = Vector3.magnitudeSquared(dx0);
+  const mag2Dx1 = Vector3.magnitudeSquared(dx1);
+
+  const dx = Vector3.normalize(mag2Dx0 > mag2Dx1 ? dx0 : dx1);
```
Comment (mikemccarty-vertex) [inline packages/viewer/src/lib/transforms/transformation-delta.ts:19]: This first `normalize()` is unnecessary.  The final `normalize()` is all that is needed.

File: packages/viewer/src/lib/transforms/transformation-delta.ts [hunk 2]
```diff
@@ -30,7 +27,7 @@ export function chooseOrthogonalVector(
  * If the normals are parallel, an axis direction based on a chosen orthogonal vector will be used
  * to compute the rotation matrix to rotate the plane 180 degrees.
  * @param normal1
- * @param normal2
+ * @param normal2r
```
Comment (mikemccarty-vertex) [inline packages/viewer/src/lib/transforms/transformation-delta.ts:30]: `r`?

**Review by mikemccarty-vertex — APPROVED** [review]

File: packages/viewer/src/lib/transforms/transformation-delta.ts [hunk 3]
```diff
@@ -1,26 +1,23 @@
...
-  const z = absNorm.z <= absNorm.x && absNorm.z <= absNorm.y ? 1.0 : 0.0;
+  const dx0 = Vector3.create(0, norm.z, -norm.y);
+  const dx1 = Vector3.create(-norm.z, 0, norm.x);
+
+  const mag2Dx0 = Vector3.magnitudeSquared(dx0);
+  const mag2Dx1 = Vector3.magnitudeSquared(dx1);
 
-  const vector = Vector3.create(x, y, z);
-  return Vector3.normalize(Vector3.cross(normal, vector));
+  const dx = mag2Dx0 > mag2Dx1 ? dx0 : dx1;
```
Comment (jdm717) [inline packages/viewer/src/lib/transforms/transformation-delta.ts:19]: Any chance we could update these names to be a bit more descriptive with what they represent? It's a bit hard with the current naming to follow what exactly is being computed here

File: packages/viewer/src/lib/transforms/transformation-delta.ts [hunk 4]
```diff
@@ -1,26 +1,23 @@
 import { Matrix4, Quaternion, Vector3 } from '@vertexvis/geometry';
 
 export const ALMOST_ONE = 0.9999;
+
 /**
- * For any single vector, there are an infinite number of potential orthogonal vectors. This function will determine
- * one orthogonal vector by crossing the provided vector with a unit vector in the positive X, Y, or Z directions.
+ * Calculates orthogonal coordinate frame with z-Vector pointing towards N
```
Comment (jdm717) [inline packages/viewer/src/lib/transforms/transformation-delta.ts:6]: Minor nit: could we update this comment to be a bit more clear? I'm not sure I fully follow what the `z-Vector` or `coordinate frame` are in the context of the SDK

**Review by jdm717 — COMMENTED** [review]
Changes LGTM! Just had some thoughts on the variable naming/method comment
